### DISCUSSION TITLE
VRP error: OverflowError: in method 'RoutingIndexManager_IndexToNode', argument 2 of type 'int64_t'

### DISCUSSION THREAD

**Discussion by Westlife1002** [google/or-tools#2689] [Routing (and legacy CP) questions]
original post: https://groups.google.com/g/or-tools-discuss/c/RplMRoTdV78

I am trying to add Pickup_Delivery capability for my model.
However, I got some issues related to manager index conversion. I have no idea what's the root cause of below error:

Traceback (most recent call last):
```sh
  File "C:/Users/usr/Dropbox/usr/VRP_17.0.py", line 273, in vehicle_cost_callback
    from_node = manager.IndexToNode(from_index)
  File "C:\Users\usr\AppData\Roaming\Python\Python36\site-packages\ortools\constraint_solver\pywrapcp.py", line 2818, in IndexToNode
    return _pywrapcp.RoutingIndexManager_IndexToNode(self, index)
OverflowError: in method 'RoutingIndexManager_IndexToNode', argument 2 of type 'int64_t'
```

This error happens in below function, which is working fine before I make adjustment for Pickup_Delivery setting:
```python
def define_total_cost(self, data, routing, manager):
     for vehicle_idx in range(data.num_vehicles):
           def vehicle_cost_callback(from_index, to_index, i=vehicle_idx):
                 from_node = manager.IndexToNode(from_index)
                 to_node = manager.IndexToNode(to_index)
                 return cost_vector()[i] * self.dist_matrix[from_node][to_node]
           cost_callback_index = routing.RegisterTransitCallback(vehicle_cost_callback)
           routing.SetArcCostEvaluatorOfVehicle(cost_callback_index, vehicle_idx)
```

Here are all my codes and  3  input csv (i.e. order_info, dist_matrix, time_matrix)
https://gist.github.com/Westlife1002/3e4cce18dba6c6f8ac530934c011d920

Great thanks for your help.

**Comment by lperron** [comment]
Most likely your input is not integral.

**Comment by Westlife1002** [comment]
> Most likely your input is not integral.

 it's same input before I add new configurations for Pickup_Delivery. It's working fine for the prior version. I guess it's something wrong related to the Pickup_Delivery related settings.

**Comment by Westlife1002** [comment]
I uploaded a wrong version of code. I have updated in my gist. It's still the same error.

**Reply by Mizux** [reply]
on my way to run it

BTW wrong line: https://gist.github.com/Westlife1002/3e4cce18dba6c6f8ac530934c011d920#file-code_vrp__addpickup_delivery-py-L147

**Reply by Mizux** [reply]
after few tests, it seems that above ~300 vehicles, solver can't found a first solution the max I manage was
```python
    if which_button_is_selected == "test":
        t1 = 36
        t2 = 36
        t3 = 36
        t4 = 36
        t5 = 36
        t6 = 36
        t7 = 36
        t8 = 36
```

Otherwise all your callback return floating point value while it is **mandatory** to return integer...

**Reply by Mizux** [reply]
see: https://gist.github.com/Mizux/3a890dd32855deb037165e6dc5dfd1e7

**Reply by Westlife1002** [reply]
Great thanks Mizux for your kind help.

I spent three whole days to debug this problem. It seems I found where the problem is. (Please use the updated code in my gist: https://gist.github.com/Westlife1002/3e4cce18dba6c6f8ac530934c011d920)

there is an unexpected error line in def distance_evaluator, caused when I delete the great circle distance calculation function since I switch to obtaining the navigation distance from a navigation system. I believe this unexpected error line is the actual problem which interrupted the model. 
![image](https://user-images.githubusercontent.com/34776946/127026237-f0f2ba11-c611-4ce8-8205-3575cedfe5d4.png)

however, the Traceback gave me a wrong information saying the error is related to the "manager.IndexToNode(from_index)" transformation, which mislead me going into completely wrong direction of debugging.

```
Traceback (most recent call last):
  File "C:/Users/usr/Dropbox/Jie/VRP_17.0.py", line 281, in vehicle_cost_callback
    from_node = manager.IndexToNode(from_index)
  File "C:\Users\usr\AppData\Roaming\Python\Python36\site-packages\ortools\constraint_solver\pywrapcp.py", line 2818, in IndexToNode
    return _pywrapcp.RoutingIndexManager_IndexToNode(self, index)
OverflowError: in method 'RoutingIndexManager_IndexToNode', argument 2 of type 'int64_t'
```
if I commented the error line above, the program runs ok. why does not the traceback point to right position of the real issue?
